evennia/utils/opclasses.py

```diff
@@ -0,0 +1,319 @@
+import datetime as _dt
+from evennia.utils.ansi import ANSIString as _ANSI
+from evennia.utils.validfuncs import _TZ_DICT
+from evennia.utils.valid import VALID_HANDLER as _VAL
+
+
+class _BaseOption(object):
```

Comment:
How do I prevent it from doing weird circular imports here? I'm admittedly a little unsure about the best way to lay out these modules and where to put things. But that's why I have you!

Reply:
If you want to avoid a specific circular import you need to put the import into the class so it's not loaded at the time of importing the module. I usually use this pattern:

```python
_NAME = None

class Foo(object):
    # ...
    def bar(self):
        global _NAME
        if not _NAME:
            from library import name as _NAME

        _NAME.method.function() ...
```

This used to carry some performance benefits I have since forgotten what they were. It's necessary in the flat API to delay-load imports to be made available at a global level. In the case of avoiding a circular import I have not explored if this kind of pattern is actually needed from a performance perspective in py3 anymore; possibly just doing the import inside the class is enough since imports should be cached anyway.